Fix: add handling of new features element in IRE Muds XML map format

SlySven opened the pull request:
These were making their maps unloadable by Mudlet as the new `<features>` element, containing one or more `<feature type="????">` were breaking the reader code.

Whilst examining the new files I also spotted that they are including some new exits of the form `<exit special="1", command="Abcde", target="12345">` whilst those are new the current reader will just silently ignore them. However it is easy to parse them as well.

Furthermore "exits" can also have `door="1"` and or `hidden="1"` *attributes* I thought I had put in code to parse those (to convert `door="1"` into an "open" door marking unless there was a `hidden="1"` to turn it into a "locked" door marking) years ago but it seems that detail never made it into the published/used code-base. 🤷‍♂️ 

This should close #6073.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: add handling of new features element in I.R.E. Muds XML map format

These were making their maps unloadable by Mudlet as the new `<features>`
element, containing one or more `<feature type="????">` were breaking the
reader code.

Whilst examining the new files I also spotted that they are including some
new exits of the form `<exit special="1", command="Abcde", target="12345">`
whilst those are new the current reader will just silently ignore them.
However it is easy to parse them as well.

Furthermore "exits" can also have `door="1"` and or `hidden="1"`
*attributes* I thought I had put in code to parse those (to convert
`door="1"` into an "open" door marking unless there was a `hidden="1"` to
turn it into a "locked" door marking) years ago but it seems that detail
never made it into the publised/used code-base. :man-shrugging:

This should close #6073.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: fix spelling error to keep a GHA bot happy

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/XMLimport.cpp:
```diff
@@ -447,12 +448,36 @@ void XMLimport::readRooms(QMultiHash<int, int>& areaRoomsHash)
+void XMLimport::readRoomFeature(TRoom* pR)
+{
+    if (Q_LIKELY(attributes().hasAttribute(qsl("type")))) {
+        pR->userData.insert(qsl("feature_%1").arg(attributes().value(qsl("type"))), qsl("true"));
```
IRE mapper uses `feature-` to lets align with that 
```suggestion
        pR->userData.insert(qsl("feature-%1").arg(attributes().value(qsl("type"))), qsl("true"));
```

SlySven replied:
> IRE mapper uses feature- to lets align with that

Whose mapper? The one we ship? Or one developed by @keneanung *et al*? Or something else entirely?

BTW are separate entries, like `feature_shop="true"`, `feature_teleporter="true"` etc. the best way to store the data in the room user data? Or would a `features={"shop"; "teleporter"}` *table* be better, note that the details are being stored as string(s) in the room user data and I am not clear if the latter would automagically get converted to a table when the Lua sub-system reads it or not?

keneanung replied:
> Whose mapper? The one we ship? Or one developed by @keneanung et al? Or something else entirely?

Those are the same :stuck_out_tongue:

We currently use `feature-shop = true` as that makes searching easier, apparently.

SlySven replied:
Okay, then, will make what this PR does the same...

vadi2 reviewed (approved):
LGTM, just the one note about features

SlySven committed:
Update src/XMLimport.cpp

Change the format of the string used as the key for the room user data.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

Co-authored-by: Vadim Peretokin <vperetokin@hey.com>
